### ISSUE TITLE
[BUG] Unknown CMake command "CPMFindPackage"

### ISSUE THREAD

**Issue by ChenZhongPu** [rapidsai/cudf#7537]
**Describe the bug**
`./build.sh` throws an error: Unknown CMake command "CPMFindPackage".

**Steps/Code to reproduce bug**
`./build.sh`

```
CMake Error at cmake/thirdparty/CUDF_GetJitify.cmake:20 (CPMFindPackage):
  Unknown CMake command "CPMFindPackage".
Call Stack (most recent call first):
  cmake/thirdparty/CUDF_GetJitify.cmake:29 (find_and_configure_jitify)
  CMakeLists.txt:126 (include)
```

**Environment overview**
 - Environment location: a Ubuntu 20.04 machine
 - Method of cuDF install: conda

**Environment details**
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 444d9f2336dd69cd9aad99ec3c676332c5e43ec2 (HEAD -> branch-0.19, tag: branch-0.19-latest, origin/branch-0.19, origin/HEAD)
     Author: David <45795991+davidwendt@users.noreply.github.com>
     Date:   Mon Mar 8 21:35:12 2021 -0500
     
     Reduce compile time/size for scan.cu (#7516)
     
     This PR reduces the number of calls to `inclusive_scan` and `exclusive_scan` by using a `null_replace_accessor` that allows non-nullable columns. This reduces the compile time and size of `scan.cu` by half. This PR also includes a scan gbenchmark that shows no change in performance from the original implementation.
     
     Authors:
     - David (@davidwendt)
     
     Approvers:
     - Paul Taylor (@trxcllnt)
     - Jake Hemstad (@jrhemstad)
     
     URL: https://github.com/rapidsai/cudf/pull/7516
     **git submodules***
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=20.04
     DISTRIB_CODENAME=focal
     DISTRIB_DESCRIPTION="Ubuntu 20.04.2 LTS"
     NAME="Ubuntu"
     VERSION="20.04.2 LTS (Focal Fossa)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 20.04.2 LTS"
     VERSION_ID="20.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=focal
     UBUNTU_CODENAME=focal
     Linux zhongpu-ubuntu 5.4.0-66-generic #74-Ubuntu SMP Wed Jan 27 22:54:38 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Tue Mar  9 12:59:02 2021
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 460.56       Driver Version: 460.56       CUDA Version: 11.2     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |                               |                      |               MIG M. |
     |===============================+======================+======================|
     |   0  GeForce RTX 3070    Off  | 00000000:01:00.0  On |                  N/A |
     |  0%   41C    P8    26W / 240W |    487MiB /  7948MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                                  |
     |  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
     |        ID   ID                                                   Usage      |
     |=============================================================================|
     |    0   N/A  N/A      1199      G   /usr/lib/xorg/Xorg                102MiB |
     |    0   N/A  N/A     18086      G   /usr/lib/xorg/Xorg                264MiB |
     |    0   N/A  N/A     18199      G   /usr/bin/gnome-shell              103MiB |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:                    x86_64
     CPU op-mode(s):                  32-bit, 64-bit
     Byte Order:                      Little Endian
     Address sizes:                   39 bits physical, 48 bits virtual
     CPU(s):                          12
     On-line CPU(s) list:             0-11
     Thread(s) per core:              2
     Core(s) per socket:              6
     Socket(s):                       1
     NUMA node(s):                    1
     Vendor ID:                       GenuineIntel
     CPU family:                      6
     Model:                           165
     Model name:                      Intel(R) Core(TM) i5-10600KF CPU @ 4.10GHz
     Stepping:                        5
     CPU MHz:                         4788.886
     CPU max MHz:                     4800.0000
     CPU min MHz:                     800.0000
     BogoMIPS:                        8199.79
     Virtualization:                  VT-x
     L1d cache:                       192 KiB
     L1i cache:                       192 KiB
     L2 cache:                        1.5 MiB
     L3 cache:                        12 MiB
     NUMA node0 CPU(s):               0-11
     Vulnerability Itlb multihit:     KVM: Vulnerable
     Vulnerability L1tf:              Not affected
     Vulnerability Mds:               Not affected
     Vulnerability Meltdown:          Not affected
     Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
     Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
     Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
     Vulnerability Srbds:             Not affected
     Vulnerability Tsx async abort:   Not affected
     Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp pku ospke md_clear flush_l1d arch_capabilities
     
     ***CMake***
     /home/zhongpu/miniconda3/envs/cudf_dev/bin/cmake
     cmake version 3.18.5
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
     Copyright (C) 2019 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda-11.2/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2021 NVIDIA Corporation
     Built on Thu_Jan_28_19:32:09_PST_2021
     Cuda compilation tools, release 11.2, V11.2.142
     Build cuda_11.2.r11.2/compiler.29558016_0
     
     ***Python***
     /home/zhongpu/miniconda3/envs/cudf_dev/bin/python
     Python 3.7.10
     
     ***Environment Variables***
     PATH                            : /home/zhongpu/miniconda3/envs/cudf_dev/bin:/home/zhongpu/miniconda3/condabin:/usr/local/cuda-11.2/bin:/home/zhongpu/miniconda3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/lib/jvm/java-11-openjdk-amd64/bin
     LD_LIBRARY_PATH                 :
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/zhongpu/miniconda3/envs/cudf_dev
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/zhongpu/miniconda3/condabin/conda
     # packages in environment at /home/zhongpu/miniconda3/envs/cudf_dev:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     _openmp_mutex             4.5                       1_gnu    conda-forge
     abseil-cpp                20200923.3           h9c3ff4c_0    conda-forge
     alabaster                 0.7.12                     py_0    conda-forge
     apipkg                    1.5                        py_0    conda-forge
     appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
     argon2-cffi               20.1.0           py37h5e8e339_2    conda-forge
     arrow-cpp                 1.0.1           py37hffab67d_32_cuda    conda-forge
     arrow-cpp-proc            3.0.0                      cuda    conda-forge
     async_generator           1.10                       py_0    conda-forge
     attrs                     20.3.0             pyhd3deb0d_0    conda-forge
     aws-c-cal                 0.4.5                hb3cee0b_7    conda-forge
     aws-c-common              0.5.1                h7f98852_0    conda-forge
     aws-c-event-stream        0.2.7                hf6e4101_0    conda-forge
     aws-c-io                  0.9.0                hd4bc808_1    conda-forge
     aws-checksums             0.1.11               hbd38b91_2    conda-forge
     aws-sdk-cpp               1.8.138              hfa2f2a4_3    conda-forge
     babel                     2.9.0              pyhd3deb0d_0    conda-forge
     backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
     backports                 1.0                        py_2    conda-forge
     backports.functools_lru_cache 1.6.1                      py_0    conda-forge
     black                     19.10b0                    py_4    conda-forge
     bleach                    3.3.0              pyh44b312d_0    conda-forge
     bokeh                     2.3.0            py37h89c1867_0    conda-forge
     boost-cpp                 1.75.0               hc6e9bd1_0    conda-forge
     brotli                    1.0.9                h9c3ff4c_4    conda-forge
     brotlipy                  0.7.0           py37h5e8e339_1001    conda-forge
     bzip2                     1.0.8                h7f98852_4    conda-forge
     c-ares                    1.17.1               h7f98852_1    conda-forge
     ca-certificates           2020.12.5            ha878542_0    conda-forge
     cachetools                4.2.1              pyhd8ed1ab_0    conda-forge
     certifi                   2020.12.5        py37h89c1867_1    conda-forge
     cffi                      1.14.5           py37hc58025e_0    conda-forge
     cfgv                      3.2.0                      py_0    conda-forge
     chardet                   4.0.0            py37h89c1867_1    conda-forge
     clang                     8.0.1                hc9558a2_2    conda-forge
     clang-tools               8.0.1                hc9558a2_2    conda-forge
     clangxx                   8.0.1                         2    conda-forge
     click                     7.1.2              pyh9f0ad1d_0    conda-forge
     cloudpickle               1.6.0                      py_0    conda-forge
     cmake                     3.18.5               h1f3970d_0    rapidsai-nightly
     cmake_setuptools          0.1.3                      py_0    rapidsai
     colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
     commonmark                0.9.1                      py_0    conda-forge
     cryptography              3.4.4            py37hf1a17b8_0    conda-forge
     cudatoolkit               11.0.221             h6bb024c_0    nvidia
     cudnn                     8.0.0                cuda11.0_0    nvidia
     cupy                      8.0.0            py37h0ce7dbb_0    rapidsai
     cython                    0.29.22          py37hcd2ae1e_0    conda-forge
     cytoolz                   0.11.0           py37h5e8e339_3    conda-forge
     dask                      2021.3.0+8.gde0e6e8c          pypi_0    pypi
     decorator                 4.4.2                      py_0    conda-forge
     defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
     distlib                   0.3.1              pyh9f0ad1d_0    conda-forge
     distributed               2021.3.0+2.g6f95c632          pypi_0    pypi
     dlpack                    0.3                  he1b5a44_1    conda-forge
     docutils                  0.16             py37h89c1867_3    conda-forge
     double-conversion         3.1.5                he1b5a44_2    conda-forge
     editdistance              0.5.3            py37hcd2ae1e_3    conda-forge
     entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
     execnet                   1.8.0              pyh44b312d_0    conda-forge
     expat                     2.2.10               h9c3ff4c_0    conda-forge
     fastavro                  1.3.2            py37h5e8e339_0    conda-forge
     fastrlock                 0.5              py37hcd2ae1e_2    conda-forge
     filelock                  3.0.12             pyh9f0ad1d_0    conda-forge
     flake8                    3.8.3                      py_1    conda-forge
     flatbuffers               1.12.0               h58526e2_0    conda-forge
     freetype                  2.10.4               h0708190_1    conda-forge
     fsspec                    0.8.7              pyhd8ed1ab_0    conda-forge
     future                    0.18.2           py37h89c1867_3    conda-forge
     gflags                    2.2.2             he1b5a44_1004    conda-forge
     glog                      0.4.0                h49b9bf7_3    conda-forge
     gmp                       6.2.1                h58526e2_0    conda-forge
     grpc-cpp                  1.36.2               h7919d58_0    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     hypothesis                6.6.0              pyhd8ed1ab_0    conda-forge
     icu                       68.1                 h58526e2_0    conda-forge
     identify                  2.1.0              pyhd8ed1ab_0    conda-forge
     idna                      2.10               pyh9f0ad1d_0    conda-forge
     imagesize                 1.2.0                      py_0    conda-forge
     importlib-metadata        3.7.2            py37h89c1867_0    conda-forge
     importlib_metadata        3.7.2                hd8ed1ab_0    conda-forge
     iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
     ipykernel                 5.5.0            py37h888b3d9_1    conda-forge
     ipython                   7.21.0           py37h888b3d9_0    conda-forge
     ipython_genutils          0.2.0                      py_1    conda-forge
     isort                     5.0.7            py37hc8dfbb8_0    conda-forge
     jedi                      0.18.0           py37h89c1867_2    conda-forge
     jinja2                    2.11.3             pyh44b312d_0    conda-forge
     jpeg                      9d                   h36c2ea0_0    conda-forge
     jsonschema                3.2.0              pyhd8ed1ab_3    conda-forge
     jupyter_client            6.1.11             pyhd8ed1ab_1    conda-forge
     jupyter_core              4.7.1            py37h89c1867_0    conda-forge
     jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
     krb5                      1.17.2               h926e7f8_0    conda-forge
     lcms2                     2.12                 hddcbb42_0    conda-forge
     ld_impl_linux-64          2.35.1               hea4e1c9_2    conda-forge
     libblas                   3.9.0                8_openblas    conda-forge
     libcblas                  3.9.0                8_openblas    conda-forge
     libcurl                   7.75.0               hc4aaa36_0    conda-forge
     libedit                   3.1.20191231         he28a2e2_2    conda-forge
     libev                     4.33                 h516909a_1    conda-forge
     libevent                  2.1.10               hcdb4288_3    conda-forge
     libffi                    3.3                  h58526e2_2    conda-forge
     libgcc-ng                 9.3.0               h2828fa1_18    conda-forge
     libgfortran-ng            9.3.0               hff62375_18    conda-forge
     libgfortran5              9.3.0               hff62375_18    conda-forge
     libgomp                   9.3.0               h2828fa1_18    conda-forge
     liblapack                 3.9.0                8_openblas    conda-forge
     libllvm10                 10.0.1               he513fc3_3    conda-forge
     libllvm8                  8.0.1                hc9558a2_0    conda-forge
     libnghttp2                1.43.0               h812cca2_0    conda-forge
     libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
     libpng                    1.6.37               h21135ba_2    conda-forge
     libprotobuf               3.15.5               h780b84a_0    conda-forge
     librmm                    0.19.0a210308   cuda11.0_g9f241bd_37    rapidsai-nightly
     libsodium                 1.0.18               h36c2ea0_1    conda-forge
     libssh2                   1.9.0                ha56f1ee_6    conda-forge
     libstdcxx-ng              9.3.0               h6de172a_18    conda-forge
     libthrift                 0.13.0               h5aa387f_6    conda-forge
     libtiff                   4.2.0                hdc55705_0    conda-forge
     libutf8proc               2.6.1                h7f98852_0    conda-forge
     libuv                     1.41.0               h7f98852_0    conda-forge
     libwebp-base              1.2.0                h7f98852_0    conda-forge
     llvmlite                  0.35.0           py37h9d7f4d0_1    conda-forge
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.9.3                h9c3ff4c_0    conda-forge
     markdown                  3.3.4              pyhd8ed1ab_0    conda-forge
     markupsafe                1.1.1            py37h5e8e339_3    conda-forge
     mccabe                    0.6.1                      py_1    conda-forge
     mimesis                   4.0.0              pyh9f0ad1d_0    conda-forge
     mistune                   0.8.4           py37h5e8e339_1003    conda-forge
     more-itertools            8.7.0              pyhd8ed1ab_0    conda-forge
     msgpack-python            1.0.2            py37h2527ec5_1    conda-forge
     mypy                      0.782                      py_0    conda-forge
     mypy_extensions           0.4.3            py37h89c1867_3    conda-forge
     nbclient                  0.5.3              pyhd8ed1ab_0    conda-forge
     nbconvert                 6.0.7            py37h89c1867_3    conda-forge
     nbformat                  5.1.2              pyhd8ed1ab_1    conda-forge
     nbsphinx                  0.8.1              pyh44b312d_0    conda-forge
     nccl                      2.7.8.1            h4962215_100    nvidia
     ncurses                   6.2                  h58526e2_4    conda-forge
     nest-asyncio              1.4.3              pyhd8ed1ab_0    conda-forge
     nodeenv                   1.5.0              pyh9f0ad1d_0    conda-forge
     notebook                  6.2.0            py37h89c1867_0    conda-forge
     numba                     0.52.0           py37hdc94413_0    conda-forge
     numpy                     1.20.1           py37haa41c4c_0    conda-forge
     numpydoc                  1.1.0                      py_1    conda-forge
     nvtx                      0.2.3            py37h5e8e339_0    conda-forge
     olefile                   0.46               pyh9f0ad1d_1    conda-forge
     openssl                   1.1.1j               h7f98852_0    conda-forge
     orc                       1.6.7                heec2584_1    conda-forge
     packaging                 20.9               pyh44b312d_0    conda-forge
     pandas                    1.2.3            py37hdc94413_0    conda-forge
     pandoc                    1.19.2                        0    conda-forge
     pandocfilters             1.4.2                      py_1    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     parso                     0.8.1              pyhd8ed1ab_0    conda-forge
     partd                     1.1.0                      py_0    conda-forge
     pathspec                  0.8.1              pyhd3deb0d_0    conda-forge
     pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
     pickleshare               0.7.5                   py_1003    conda-forge
     pillow                    8.1.2            py37h4600e1f_0    conda-forge
     pip                       21.0.1             pyhd8ed1ab_0    conda-forge
     pluggy                    0.13.1           py37h89c1867_4    conda-forge
     pre-commit                2.11.0           py37h89c1867_0    conda-forge
     pre_commit                2.11.0               hd8ed1ab_0    conda-forge
     prometheus_client         0.9.0              pyhd3deb0d_0    conda-forge
     prompt-toolkit            3.0.16             pyha770c72_0    conda-forge
     protobuf                  3.15.5           py37hcd2ae1e_0    conda-forge
     psutil                    5.8.0            py37h5e8e339_1    conda-forge
     ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
     py                        1.10.0             pyhd3deb0d_0    conda-forge
     py-cpuinfo                7.0.0              pyh9f0ad1d_0    conda-forge
     pyarrow                   1.0.1           py37h3dc597d_32_cuda    conda-forge
     pycodestyle               2.6.0              pyh9f0ad1d_0    conda-forge
     pycparser                 2.20               pyh9f0ad1d_2    conda-forge
     pyflakes                  2.2.0              pyh9f0ad1d_0    conda-forge
     pygments                  2.8.1              pyhd8ed1ab_0    conda-forge
     pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
     pyorc                     0.4.0                    pypi_0    pypi
     pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
     pyrsistent                0.17.3           py37h5e8e339_2    conda-forge
     pysocks                   1.7.1            py37h89c1867_3    conda-forge
     pytest                    6.2.2            py37h89c1867_0    conda-forge
     pytest-benchmark          3.2.3              pyh9f0ad1d_0    conda-forge
     pytest-forked             1.3.0              pyhd3deb0d_0    conda-forge
     pytest-xdist              2.2.1              pyhd8ed1ab_0    conda-forge
     python                    3.7.10          hffdb5ce_100_cpython    conda-forge
     python-dateutil           2.8.1                      py_0    conda-forge
     python_abi                3.7                     1_cp37m    conda-forge
     pytz                      2021.1             pyhd8ed1ab_0    conda-forge
     pyyaml                    5.4.1            py37h5e8e339_0    conda-forge
     pyzmq                     22.0.3           py37h336d617_1    conda-forge
     rapidjson                 1.1.0             he1b5a44_1002    conda-forge
     re2                       2020.11.01           h58526e2_0    conda-forge
     readline                  8.0                  he28a2e2_2    conda-forge
     recommonmark              0.7.1              pyhd8ed1ab_0    conda-forge
     regex                     2020.11.13       py37h5e8e339_1    conda-forge
     requests                  2.25.1             pyhd3deb0d_0    conda-forge
     rhash                     1.4.1                h7f98852_0    conda-forge
     rmm                       0.19.0a210308   cuda_11.0_py37_g9f241bd_37    rapidsai-nightly
     s2n                       1.0.0                h9b69904_0    conda-forge
     send2trash                1.5.0                      py_0    conda-forge
     setuptools                49.6.0           py37h89c1867_3    conda-forge
     six                       1.15.0             pyh9f0ad1d_0    conda-forge
     snappy                    1.1.8                he1b5a44_3    conda-forge
     snowballstemmer           2.1.0              pyhd8ed1ab_0    conda-forge
     sortedcontainers          2.3.0              pyhd8ed1ab_0    conda-forge
     spdlog                    1.7.0                hc9558a2_2    conda-forge
     sphinx                    3.5.2              pyhd8ed1ab_0    conda-forge
     sphinx-copybutton         0.3.1              pyhd8ed1ab_0    conda-forge
     sphinx-markdown-tables    0.0.15             pyhd3deb0d_0    conda-forge
     sphinx_rtd_theme          0.5.1              pyhd3deb0d_0    conda-forge
     sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
     sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
     sphinxcontrib-htmlhelp    1.0.3                      py_0    conda-forge
     sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
     sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
     sphinxcontrib-serializinghtml 1.1.4                      py_0    conda-forge
     sphinxcontrib-websupport  1.2.4              pyh9f0ad1d_0    conda-forge
     sqlite                    3.34.0               h74cdb3f_0    conda-forge
     streamz                   0.6.2              pyh44b312d_0    conda-forge
     tblib                     1.6.0                      py_0    conda-forge
     terminado                 0.9.2            py37h89c1867_0    conda-forge
     testpath                  0.4.4                      py_0    conda-forge
     tk                        8.6.10               h21135ba_1    conda-forge
     toml                      0.10.2             pyhd8ed1ab_0    conda-forge
     toolz                     0.11.1                     py_0    conda-forge
     tornado                   6.1              py37h5e8e339_1    conda-forge
     traitlets                 5.0.5                      py_0    conda-forge
     typed-ast                 1.4.2            py37h5e8e339_0    conda-forge
     typing_extensions         3.7.4.3                    py_0    conda-forge
     urllib3                   1.26.3             pyhd8ed1ab_0    conda-forge
     virtualenv                20.4.2           py37h89c1867_0    conda-forge
     wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
     webencodings              0.5.1                      py_1    conda-forge
     wheel                     0.36.2             pyhd3deb0d_0    conda-forge
     xz                        5.2.5                h516909a_1    conda-forge
     yaml                      0.2.5                h516909a_0    conda-forge
     zeromq                    4.3.4                h9c3ff4c_0    conda-forge
     zict                      2.0.0                      py_0    conda-forge
     zipp                      3.4.1              pyhd8ed1ab_0    conda-forge
     zlib                      1.2.11            h516909a_1010    conda-forge
     zstd                      1.4.9                ha95c52a_0    conda-forge
     
</pre></details>

**Comment by kkraus14** [comment]
cc @robertmaynard @trxcllnt any ideas here? Environment has CMake 3.18.5

**Comment by robertmaynard** [comment]
`home/zhongpu/miniconda3/envs/cudf_dev/bin/cmake`

Is this a custom build of CMake? My first thought was that SSL wasn't enabled and therefore CPM can't be downloaded as  CMake doesn't have https support.

**Comment by robertmaynard** [comment]
> > `home/zhongpu/miniconda3/envs/cudf_dev/bin/cmake`
> > Is this a custom build of CMake? My first thought was that SSL wasn't enabled and therefore CPM can't be downloaded as  CMake doesn't have https support.
> 
> It's the CMake conda package

If the download failed due to something like a `500` timeout error we would have this problem.
I think we should move over to `FetchContent` to get the CPM file, as it automatically does download retries ( with backoff ), and provides a nice API to verify that the downloaded content is correct.

**Comment by kkraus14** [comment]
> If the download failed due to something like a `500` timeout error we would have this problem.
> I think we should move over to `FetchContent` to get the CPM file, as it automatically does download retries ( with backoff ), and provides a nice API to verify that the downloaded content is correct.

We should probably contribute this upstream to CPM as well. It also allows us to use a hash for the download to guarantee the package doesn't get changed out from underneath us.

**Comment by kkraus14** [comment]
@ChenZhongPu thanks for reporting. Does this happen consistently for you?

**Comment by trxcllnt** [comment]
@ChenZhongPu Is `raw.githubusercontent.com` blocked on your network? Can you access [the file](https://raw.githubusercontent.com/TheLartians/CPM.cmake/3b404296b539e596f39421c4e92bc803b299d964/cmake/CPM.cmake) in the browser?

**Comment by kkraus14** [comment]
That should have been fixed in #7533, so maybe try pulling the latest and trying again?

**Comment by trxcllnt** [comment]
@ChenZhongPu that looks like an out of memory error during compilation. Can try running with fewer concurrent jobs via `PARALLEL_LEVEL=2 ./build.sh`

**Comment by ChenZhongPu** [comment]
> @ChenZhongPu that looks like an out of memory error during compilation. Can try running with fewer concurrent jobs via `PARALLEL_LEVEL=2 ./build.sh`

Thanks a lot. It works.

**Comment by chenrui17** [comment]
i met a similar error when i build libcudf use `scl enable devtoolset-7 "java/ci/build-in-docker.sh"`, and i set the PARALLEL_LEVEL=2 ,but it also not works.

**Comment by kkraus14** [comment]
> i met a similar error when i build libcudf use `scl enable devtoolset-7 "java/ci/build-in-docker.sh"`, and i set the PARALLEL_LEVEL=2 ,but it also not works.

Could you please open a new issue for discussion? If you're killed `Error 2` it's almost always that compilation ran out of memory.
